### PR TITLE
Use registry domain for k8s-api-healthz and wait for domains script.

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/7773

Fixes last 2 instances of `quay.io` hardcoding in `v_5_0_0`. This is so we use the retagged images in China.

### CODE REVIEW

File: v_5_0_0/files/manifests/k8s-api-healthz.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ spec:
       command:
         - /k8s-api-healthz
         - --api-endpoint="https://$(HOST_IP):443/healthz"
-      image: quay.io/giantswarm/k8s-api-healthz:1c0cdf1ed5ee18fdf59063ecdd84bf3787f80fac
+      image:  {{ .RegistryDomain }}/giantswarm/k8s-api-healthz:1c0cdf1ed5ee18fdf59063ecdd84bf3787f80fac
```
Comment (calvix) [inline v_5_0_0/files/manifests/k8s-api-healthz.yaml]: we need to bump the version as only the latest tag is in china

Comment (calvix) [reply v_5_0_0/files/manifests/k8s-api-healthz.yaml]: `0999549a4c334b646288d08bd2c781c6aae2e12f` should be the last tag and should be also in china

Comment (rossf7) [reply v_5_0_0/files/manifests/k8s-api-healthz.yaml]: @calvix Thanks. Done and confirmed here.  https://circleci.com/gh/giantswarm/k8s-api-healthz/40

**Comment by rossf7** [comment]
Tested on `giraffe`.

**Pull request merged**